### DISCUSSION TITLE
SpatialPad, ResizeWithPadOrCrop fails in simple example

### DISCUSSION THREAD

**Discussion by juanprietob** [Project-MONAI/MONAI#6304] [Q&A] [answered]
**Describe the bug**
When executing SpatialPad or ResizeWithPadOrCrop transform an exception is thrown

`ValueError: Sequence must have length 2, got 3.`

**To Reproduce**
```
from monai import transforms
import numpy as np 

image = np.random.randint(0, 256, (50, 50, 50), dtype=np.uint8)
# Define the desired spatial dimensions after padding
spatial_size = (64, 64, 64)
# Create a SpatialPad instance
spatial_pad = transforms.SpatialPad(spatial_size=spatial_size, mode='constant')
# Apply the padding transform to the image
padded_image = spatial_pad(image)
print("Original image shape:", image.shape)
print("Padded image shape:", padded_image.shape)
```

**Expected behavior**
The image is padded to the desire dimension

**Screenshots**

![image](https://user-images.githubusercontent.com/7086191/230113676-5af2587b-101f-4703-8234-95728361c82e.png)

**Environment**

Ensuring you use the relevant python executable, please paste the output of:
Python 3.8.12
monai 1.1.0

```
python -c 'import monai; monai.config.print_debug_info()'
```

================================
Printing MONAI config...
================================
MONAI version: 1.1.0
Numpy version: 1.23.1
Pytorch version: 1.12.1+cu113
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: a2ec3752f54bfc3b40e7952234fbeb5452ed63e3
MONAI __file__: /mnt/raid/home/jprieto/anaconda3/envs/torch_us/lib/python3.8/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: NOT INSTALLED or UNKNOWN VERSION.
Nibabel version: NOT INSTALLED or UNKNOWN VERSION.
scikit-image version: NOT INSTALLED or UNKNOWN VERSION.
Pillow version: 9.2.0
Tensorboard version: 2.12.0
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.13.1+cu113
tqdm version: 4.64.0
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: 5.9.4
pandas version: 1.4.3
einops version: 0.6.0
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: 0.4.3

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

================================
Printing system config...
================================
System: Linux
Linux version: Red Hat Enterprise Linux 8.7 (Ootpa)
Platform: Linux-4.18.0-372.32.1.el8_6.x86_64-x86_64-with-glibc2.17
Processor: x86_64
Machine: x86_64
Python version: 3.8.12
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: []
Num physical CPUs: 48
Num logical CPUs: 96
Num usable CPUs: 96
CPU usage (%): [7.9, 7.1, 7.9, 1.6, 0.0, 7.9, 7.9, 7.8, 7.9, 7.9, 0.0, 7.9, 7.9, 0.0, 7.9, 0.0, 7.9, 10.2, 0.8, 7.9, 7.1, 7.9, 8.6, 7.9, 0.0, 0.0, 0.0, 0.0, 0.0, 7.9, 7.9, 0.0, 0.0, 8.6, 7.1, 7.9, 0.0, 7.9, 8.6, 8.6, 7.9, 0.0, 7.1, 8.6, 7.8, 7.9, 7.1, 99.2, 8.6, 7.9, 7.9, 7.1, 8.6, 7.9, 0.0, 0.0, 0.0, 0.0, 7.1, 7.9, 7.9, 7.9, 0.0, 7.8, 0.0, 0.0, 8.6, 8.6, 8.6, 0.8, 0.0, 0.0, 7.9, 7.8, 7.9, 7.1, 8.6, 7.9, 0.0, 7.1, 7.9, 0.0, 0.0, 0.0, 8.6, 0.0, 0.0, 7.1, 0.0, 7.1, 7.1, 7.9, 7.8, 7.9, 7.9, 8.6]
CPU freq. (MHz): 2655
Load avg. in last 1, 5, 15 mins (%): [0.2, 0.1, 0.0]
Disk usage (%): 61.8
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 503.2
Available memory (GB): 435.8
Used memory (GB): 63.1

================================
Printing GPU config...
================================
Num GPUs: 4
Has CUDA: True
CUDA version: 11.3
cuDNN enabled: True
cuDNN version: 8302
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80', 'sm_86']
GPU 0 Name: NVIDIA A100-SXM-80GB
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 108
GPU 0 Total memory (GB): 79.3
GPU 0 CUDA capability (maj.min): 8.0
GPU 1 Name: NVIDIA A100-SXM-80GB
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 108
GPU 1 Total memory (GB): 79.3
GPU 1 CUDA capability (maj.min): 8.0
GPU 2 Name: NVIDIA A100-SXM-80GB
GPU 2 Is integrated: False
GPU 2 Is multi GPU board: False
GPU 2 Multi processor count: 108
GPU 2 Total memory (GB): 79.3
GPU 2 CUDA capability (maj.min): 8.0
GPU 3 Name: NVIDIA A100-SXM-80GB
GPU 3 Is integrated: False
GPU 3 Is multi GPU board: False
GPU 3 Multi processor count: 108
GPU 3 Total memory (GB): 79.3
GPU 3 CUDA capability (maj.min): 8.0

**Additional context**

**Comment by KumoLiu** [comment, marked as answer]
Hi @juanprietob, Since most of the transforms in MONAI assume that the shape of the data is BxCxHxWx[D], you should add channel dim using `EnsureChannelFirst`.
Try something like:
```image = np.random.randint(0, 256, (50, 50, 50), dtype=np.uint8)[None]```.

Hope it can help you, thanks!
